core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1872,6 +1872,46 @@ class WorkspaceService(protected val userInfo: UserInfo,
     }
   }
 
+  def adminListWorkspacesWithInvalidEntityCaches: Future[PerRequestMessage] = {
```

Comment:
It looks like we're making 2 distinct database calls here.. getting the invalid caches then hydrating those ids to workspaces.  I know it would involve coding up the sql, but I'm sure it would run faster if we made it one query and I don't know the particulars of what slick does for listByIds but if it's a where in (ids) you eliminate the minuscule risk of overflowing the where in clause.